backend/substrapp/ledger_utils.py

```diff
@@ -215,34 +217,92 @@ def invoke_ledger(fcn, args=None, cc_pattern=None, sync=False, only_pkhash=True)
         return response
 
 
+@retry_on_error(exceptions=[LedgerTimeout])
+def query_ledger(fcn, args=None):
+    # careful, passing invoke parameters to query_ledger will NOT fail
+    return call_ledger('query', fcn=fcn, args=args)
+
+
+@retry_on_error(exceptions=[LedgerTimeout])
+def invoke_ledger(*args, **kwargs):
+    return _invoke_ledger(*args, **kwargs)
+
+
 @retry_on_error()
+def update_ledger(*args, **kwargs):
+    return _invoke_ledger(*args, **kwargs)
+
+
+def query_tuples(tuple_type, data_owner):
+    data = query_ledger(
+        fcn="queryFilter",
+        args={
+            'indexName': f'{tuple_type}~worker~status',
+            'attributes': f'{data_owner},todo'
+        }
+    )
+
+    data = [] if data is None else data
+
+    return data
+
+
 def get_object_from_ledger(pk, query):
     return query_ledger(fcn=query, args={'key': pk})
 
 
-@retry_on_error()
+def _wait_until_status_after_timeout(tuple_type, tuple_key, expected_status):
+    query_fcns = {
+        'traintuple': 'queryTraintuple',
+        'testtuple': 'queryTesttuple',
+        'compositeTraintuple': 'queryCompositeTraintuple',
+        'aggregatetuple': 'queryAggregatetuple',
+    }
+    query_fcn = query_fcns[tuple_type]
+
+    max_tries = 5
```

Comment:
could you use settings for those values?

Reply:
done